File: html/old/backup/sacor2_2008/parksides.java
// SACO Second Round 2008
// Problem: Parkside's Triangle
// Author:  Marco Gallotta

import java.io.*;
import java.util.Scanner;
public class parksides {
  public static void main(String args[]) throws IOException {
    Scanner in = new Scanner(System.in);
    System.out.print("Enter N: ");
    int N = in.nextInt();
    System.out.print("Enter S: ");
    int S = in.nextInt();
    int tri[][] = new int[N][N];
    for (int i = 0; i < N; i++)
      for (int j = 0; j <= i; j++) {
        tri[j][i] = S;
        S++;
        if (S == 10)
          S = 1;
      }
    for (int i = 0; i < N; i++) {
      for (int j = 0; j < i; j++)
        System.out.print("  ");
      for (int j = i; j < N; j++)
        System.out.print(tri[i][j] + " ");
      System.out.println();
    }
  }
}

File: html/old/backup/sacor2_2008/parksides.cpp
// SACO Second Round 2008
// Problem: Parkside's Triangle
// Author: Timothy Stranex

#include <iostream>
using namespace std;

int main()
{
  int N, S;
  cout << "Enter N: ";
  cin >> N;
  cout << "Enter S: ";
  cin >> S;

  for (int i=0; i<N; i++) {
    int pos = S + i + 8;
    for (int j=0; j<i; j++) {
      pos += j+1;
      cout << "  ";
    }
    for (int j=i; j<N; j++) {
      cout << pos%9+1 << " ";
      pos += j+1;
    }
    cout << endl;
  }

  return 0;
}
